src/passes/externalContractHandler/externalContractInterfaceInserter.ts

```diff
@@ -0,0 +1,154 @@
+import assert = require('assert');
+import {
+  ContractDefinition,
+  ContractKind,
+  FunctionCall,
+  FunctionDefinition,
+  FunctionKind,
+  getNodeType,
+  Identifier,
+  MemberAccess,
+  SourceUnit,
+  UserDefinedType,
+  UserDefinedTypeName,
+  VariableDeclaration,
+} from 'solc-typed-ast';
+
+import { AST } from '../../ast/ast';
+import { CairoContract } from '../../ast/cairoNodes';
+import { ASTMapper } from '../../ast/mapper';
+import { cloneASTNode } from '../../utils/cloning';
+
+function genContractInterface(
+  contract: ContractDefinition,
+  sourceUnit: SourceUnit,
+  ast: AST,
+): ContractDefinition {
+  const contractId = ast.reserveId();
+  const contractInterface = new CairoContract(
+    contractId,
+    '',
+    `${contract.name}@interface`,
```

Comment:
I'm assuming this is to prevent redaclaration of existing contracts, is there any other reason for modifying the name like this?

Reply:
Yes, it is just the avoid conflict with the existing contracts

Reply:
It is clashing

Reply:
As discussed, fixing the clashing without mangling the names like this would be more hassle than it's worth